pyface/base_application.py

```diff
@@ -0,0 +1,286 @@
+# Copyright (c) 2014-2016 by Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in enthought/LICENSE.txt and may be redistributed only
+# under the conditions described in the aforementioned license.  The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+# Thanks for using Enthought open source!
+
+""" Define a base application class to create the event loop, and launch
+the creation of application windows.
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import os
+import logging
+import sys
+
+from traits.api import Bool, Directory, Event, HasStrictTraits, Instance, Str
+
+
+logger = logging.getLogger(__name__)
+
+
+class ApplicationEvent(HasStrictTraits):
+    """ An event associated with an application """
+
+    #: The application that the event happened to.
+    application = Instance('BaseApplication')
+
+    #: The type of application event.
+    event_type = Str
+
+
+class BaseApplication(HasStrictTraits):
+    """ A base class for applications.
+
+    This class handles the basic lifecycle of an application and a few
+    fundamental facilities.  It is suitable as a base for any application,
+    not just GUI applications.
+    """
+
+    # -------------------------------------------------------------------------
+    # 'BaseApplication' interface
+    # -------------------------------------------------------------------------
+
+    # Branding ----------------------------------------------------------------
+
+    #: Human-readable application name
+    name = Str('Default Application')
+
+    #: Human-readable company name
+    company = Str('Enthought')
```

Comment:
Personally, I would remove `Enthought` and use a more neutral default like for your `name`.